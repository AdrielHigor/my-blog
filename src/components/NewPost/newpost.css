header#Newpost{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 100%;
    max-width: 450px;
    margin: 30px auto 10px;
}

header#Newpost a {
    background-color: #25282a;
    padding: 8px 18px 8px;
    border-radius: 4px;
    color: #ffffff;
    text-decoration: none;
    font-size: 15px;
}

form#Newpost {
    width: 100%;
    max-width: 680px;
    margin: 0px auto 0px;
    padding: 30px;
    border: 1px solid #25282a;
    border-radius: 4px;
    display: flex;
    flex-direction: column;
}

form#Newpost input{
    margin-bottom: 15px;
    height: 40px;
    border-radius: 4px;
    border: 1px solid #ddd;
    padding: 0 20px;
    font-size: 14px;
}

form#Newpost textarea{
    margin-bottom: 15px;
    border-radius: 4px;
    border: 1px solid #ddd;
    padding: 0 20px;
    font-size: 14px;
}

form#Newpost button{
    padding:10px 20px;
    border-radius: 4px;
    background: #25282a;
    color: #ffffff;
    font-size: 18px;
    font-weight: bold;
    cursor: pointer;
}