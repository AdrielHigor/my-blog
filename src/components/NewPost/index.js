import React, { useEffect } from "react";
import { Link, withRouter } from "react-router-dom";
import "./newpost.css";
import firebase from "../../firebase";

export default withRouter(function NewPost(props) {
    const [description, setDescription] = React.useState();
    const [image, setImage] = React.useState();
    const [title, setTitle] = React.useState();
    const [author, setAuthor] = React.useState();

    const newpost = React.useCallback(e => {
        e.preventDefault();

        if (author !== "" && title !== "" && description !== "" && image !== "") {
            firebase.makePost(author, title, image, description);
            props.history.push("/dashboard");
        } else {
            alert("Todos os campos são obrigatórios");
        }
    }, [author, description, image, props.history, title]
    )

    useEffect(() => {
        if (!firebase.getCurrent()) {
            props.history.replace("/login");
        }
        firebase.getUserInfo(info => {
            setAuthor(info.val().name);
            if (info.val().role !== "admin") {
                props.history.push("/dashboard");
            }
        });
    });

    return (
        <div>
            <header id="Newpost">
                <Link to="/dashboard">Voltar</Link>
            </header>
            <form onSubmit={newpost} id="Newpost">
                <label>Título: </label>
                <input
                    type="text"
                    placeholder="Titulo do Post"
                    value={title}
                    onChange={e => setTitle(e.target.value)}
                    autoFocus
                />
                <label>Imagem: </label>
                <input
                    type="text"
                    placeholder="Url da capa"
                    value={image}
                    onChange={e => setImage(e.target.value)}
                />
                <label>Descrição: </label>
                <textarea
                    type="text"
                    rows="10"
                    placeholder="Descrição do post"
                    value={description}
                    onChange={e => setDescription(e.target.value)}
                />
                <button type="submit">Enviar</button>
            </form>
        </div>
    );
});
