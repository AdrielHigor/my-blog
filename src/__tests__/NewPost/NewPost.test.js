import React from "react";
import NewPost from "../../components/NewPost";
import { shallow } from "enzyme";
import "../../setupTests";

describe("NewPost Test Suite", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    })

    test("NewPost rendering correctly", () => {
        const component = shallow(<NewPost.WrappedComponent />);
        expect(component).toMatchSnapshot();
    })

    test("NewPost Submit form correctly", () => {
        let component

        jest
            .spyOn(React, 'useState')
            .mockImplementationOnce(() => ["test description", jest.fn()])
            .mockImplementationOnce(() => ["http://testimage.com/teste_image.jpg", jest.fn()])
            .mockImplementationOnce(() => ["test", jest.fn()])
            .mockImplementationOnce(() => ["Test User", jest.fn()])

        const mockRegister = jest
            .spyOn(React, 'useCallback')
            .mockImplementation(() => { })

        component = shallow(<NewPost.WrappedComponent />);

        component.find("button").simulate("click");

        expect(mockRegister).toHaveBeenCalled();
    })

    test("NewPost Submit form incorrectly", () => {
        let component

        const mockRegister = jest
            .spyOn(React, 'useCallback')
            .mockImplementation(() => {
                alert("Todos os campos são obrigatórios");
            })

        const mockAlert = jest.spyOn(window, 'alert').mockImplementation(() => { });

        component = shallow(<NewPost.WrappedComponent />);

        component.find("button").simulate("click");

        expect(mockRegister).toHaveBeenCalled();
        expect(mockAlert).toHaveBeenCalled();
    })
})