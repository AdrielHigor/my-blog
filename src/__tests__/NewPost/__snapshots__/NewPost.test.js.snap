// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPost Test Suite NewPost rendering correctly 1`] = `ShallowWrapper {}`;

exports[`NewPost rendering correctly 1`] = `ShallowWrapper {}`;
